Kconfig.zephyr

```diff
@@ -911,3 +911,15 @@ config COMPAT_INCLUDES
 	  deprecated header files.
 
 endmenu
+
+config LEGACY_GENERATED_INCLUDE_PATH
+	bool "Allow for the legacy include path for generated headers (without the zephyr/ prefix)"
+	default y
+	select DEPRECATED
```

Comment:
I'd avoid `select DEPRECATED` in a first iteration, some builds enable `WARN_DEPRECATED`.